src/web/rustdoc.rs

```diff
@@ -188,16 +186,20 @@ pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
     let (target_name, has_docs): (String, bool) = {
         let rows = ctry!(conn.query(
             "SELECT target_name, rustdoc_status
-                                     FROM releases
-                                     WHERE releases.id = $1",
+             FROM releases
+             WHERE releases.id = $1",
             &[&id]
         ));
 
         (rows.get(0).get(0), rows.get(0).get(1))
     };
 
+    if target == Some("index.html") || target == Some(&target_name) {
+        target = None;
+    }
+
     if has_docs {
-        redirect_to_doc(req, &crate_name, &version, &target_name)
+        redirect_to_doc(req, &crate_name, &version, target, &target_name)
```

Comment:
At some point we should probably change this `target_name` to `crate_name` or similar ... but that can wait for another PR.

Reply:
Maybe `library_name` since non-library-target docs aren't built. I was thinking about it a little and target is actually the correct nomenclature for this: [_Cargo Targets_](https://doc.rust-lang.org/nightly/cargo/reference/cargo-targets.html), it's just unfortunate that there is overlap between platform-target and library-target.